hydrolib/core/io/ini/serializer.py

```diff
@@ -0,0 +1,245 @@
+from dataclasses import dataclass
+from functools import reduce
+from hydrolib.core.io.ini.models import CommentBlock, Document, Property, Section
+from hydrolib.core.basemodel import BaseModel
+from itertools import chain, count, repeat
+from pathlib import Path
+from typing import Any, Iterable, List, Optional, Sequence, Tuple, Union
+
+
+class SerializerConfig(BaseModel):
+    """SerializerConfig defines the configuration options of the Serializer
+
+    Attributes:
+        section_indent (int):
+            The number of spaces with which whole sections should be indented.
+            Defaults to 0.
+        property_indent (int):
+            The number of spaces with which properties should be indented relative to
+            the section header (i.e. the full indent equals the section_indent plus
+            property_indent). Defaults to 4.
+        datablock_indent (int):
+            The number of spaces with which datablock rows are indented relative to
+            the section header (i.e. the full indent equals the section_indent plus
+            datablock_indent). Defaults to 8.
+        datablock_spacing (int):
+            The number of spaces between datablock columns. Note that there might be
+            additional offset to ensure . is lined out. Defaults to 4.
+        comment_delimiter (str):
+            The character used to delimit comments. Defaults to '#'.
+    """
+
+    section_indent: int = 0
+    property_indent: int = 4
+    datablock_indent: int = 8
+    datablock_spacing: int = 4
+    comment_delimiter: str = "#"
+
+    @property
+    def total_property_indent(self) -> int:
+        return self.section_indent + self.property_indent
+
+    @property
+    def total_datablock_indent(self) -> int:
+        return self.section_indent + self.datablock_indent
+
+
+@dataclass
```

Comment:
Everything uses pydantic, and now suddenly you use a dataclass? I'm not against it, it's just weird.

Reply:
I had it originally as a simple data transfer object within the Serializer itself, but it made more sense to move it out when I added the from_section logic. You are right though that for consistency it makes more sense to just use pydantic, unless there is a good reason not too. I will change it.